### PR TITLE
build C executables

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This depends on #203 will rebase once that is merged.

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
rebased

File: Tests/Functional/TestClangModules.swift [hunk 1]
```diff
@@ -63,6 +63,13 @@ class TestClangModulesTestCase: XCTestCase {
             XCTAssertDirectoryExists(prefix, "Bar/Packages/Foo-1.2.3")
         }
     }
+    
+    func testCExecutable() {
+        fixture(name: "ValidLayouts/SingleModule/CExecutable") { prefix in
+            XCTAssertBuilds(prefix)
+            XCTAssertFileExists(prefix, ".build/debug/CExecutable")
```
Comment (mxcl) [inline Tests/Functional/TestClangModules.swift]: We should test for a known output from the executable.

Comment (aciidb0mb3r) [inline Tests/Functional/TestClangModules.swift]: done

**Comment by mxcl** [comment]
@swift-ci Please test

**Comment by aciidb0mb3r** [comment]
I guess CI will fail because of the two failing tests  ._.  let me try to fix them

**Comment by aciidb0mb3r** [comment]
All green, @mxcl can I merge 😍

**Comment by mxcl** [comment]
Looks good to me. 👍🏻

**Pull request merged**